src/trio/_tests/test_testing_raisesgroup.py

```diff
@@ -107,6 +133,96 @@ def test_flatten_subgroups() -> None:
     ):
         RaisesGroup(RaisesGroup(ValueError), flatten_subgroups=True)  # type: ignore[call-overload]
 
+    # flatten_subgroups is not sufficient to catch fully unwrapped
+    with pytest.raises(
+        AssertionError,
+        match=wrap_escape(
+            "Raised exception group did not match: ValueError() is not an exception group, but would match with `allow_unwrapped=True`",
+        ),
+    ):
+        with RaisesGroup(ValueError, flatten_subgroups=True):
+            raise ValueError
+    with pytest.raises(
+        AssertionError,
+        match=wrap_escape(
+            "Raised exception group did not match: RaisesGroup(ValueError, flatten_subgroups=True): ValueError() is not an exception group, but would match with `allow_unwrapped=True`",
+        ),
+    ):
+        with RaisesGroup(RaisesGroup(ValueError, flatten_subgroups=True)):
+            raise ExceptionGroup("", (ValueError(),))
+
+    # helpful suggestion if flatten_subgroups would make it pass
+    # TODO: this one is overly verbose - in the case of a single [unmatched] raised exception we don't need a table.
+    # TODO: Printing Exceptiongroup('', [ValueError(), TypeError()]) three times is also a different issue
+    with pytest.raises(
+        AssertionError,
+        match=wrap_escape(
+            "Raised exception group did not match: \n"
+            "The following expected exceptions did not find a match: [<class 'ValueError'>, <class 'TypeError'>]\n"
+            "The following raised exceptions did not find a match\n"
+            "  ExceptionGroup('', [ValueError(), TypeError()]):\n"
+            "    ExceptionGroup('', [ValueError(), TypeError()]) is not of type 'ValueError'\n"
+            "    ExceptionGroup('', [ValueError(), TypeError()]) is not of type 'TypeError'\n"
```

Comment:
I'm fine with printing a table here, but we could just remove the repetition:
```suggestion
            "  ExceptionGroup('', [ValueError(), TypeError()]):\n"
            "    -> not of type 'ValueError'\n"
            "    -> not of type 'TypeError'\n"
```
(particularly because I expect most reprs to be considerably longer than this)

Reply:
This turned out to be very messy, so I instead ended up printing `type(e).__name__` on a type mismatch, and not print the repr if `check` failed, and I think that ultimately resolved it cleanly. There's some non-nested simple cases that now doesn't print the repr at all, but I think they should all be fine.